src/dsp_tools/utils/xml_validation.py

```diff
@@ -119,3 +125,74 @@ def _find_xml_tags_in_simple_text_elements(
         err_msg += list_separator + list_separator.join(resources_with_illegal_xml_tags)
         return False, err_msg
     return True, ""
+
+
+def _find_mixed_encodings_in_one_text_prop(
+    xml_no_namespace: etree._Element,
+) -> tuple[bool, str]:
+    problems = check_if_only_one_encoding_is_used_per_prop_in_root(xml_no_namespace)
+    if not problems:
+        return True, ""
+    msg, df = InconsistentTextValueEncodings(problems).execute_problem_protocol()
+    if df is not None:
+        csv_path = Path(f"XML_syntax_errors_{datetime.now().strftime('%Y-%m-%d_%H%M%S')}.csv")
+        msg = f"\nAll the problems are listed in the file: '{csv_path.absolute()}'" + msg
+        df.to_csv(csv_path)
+    return False, msg
+
+
+def check_if_only_one_encoding_is_used_per_prop_in_root(
+    root: etree._Element,
+) -> list[TextValueData]:
+    """
+    Check if all the encodings in the <text> elements are consistent within one <text-prop>
+
+    This is correct:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text encoding="utf8">Text 1</text>
+        <text encoding="utf8">Text 2</text>
+    </text-prop>
+    ```
+
+    This is wrong:
+    ```
+    <text-prop name=":hasSimpleText">
+        <text encoding="utf8">Text 1</text>
+        <text encoding="xml">Text 2</text>
+    </text-prop>
+    ```
+
+    Args:
+        root: root of the data xml document
+
+    Returns:
+          True and None if all the elements are consistent
+          False and a list of all the inconsistent <text-props>
+    """
+    text_props = _get_all_ids_and_encodings_from_root(root)
+    return _check_only_one_valid_encoding_used_all_props(text_props)
+
+
+def _get_all_ids_and_encodings_from_root(
+    root: etree._Element,
+) -> list[TextValueData]:
+    res_list = []
+    for res_input in root.iterchildren(tag="resource"):
+        res_list.extend(_get_encodings_from_one_resource(res_input))
+    return res_list
+
+
+def _get_encodings_from_one_resource(resource: etree._Element) -> list[TextValueData]:
+    res_id = resource.attrib["id"]
+    return [_get_encodings_from_one_property(res_id, child) for child in list(resource.iterchildren(tag="text-prop"))]
+
+
+def _get_encodings_from_one_property(res_id: str, prop: etree._Element) -> TextValueData:
+    prop_name = prop.attrib["name"]
+    encodings = {x.attrib["encoding"] for x in prop.iterchildren()}
+    return TextValueData(res_id, prop_name, encodings)
+
+
+def _check_only_one_valid_encoding_used_all_props(text_props: list[TextValueData]) -> list[TextValueData]:
```

Comment:
I have troubles understanding the function name. And when I compare it to what the function actually does, I have even more question marks. 

If I look at the code, I would say something like: 

- "find illegal text props"
- "find text props that use multiple encodings"